main.tf

```diff
@@ -67,6 +67,7 @@ resource "aws_launch_configuration" "bastion" {
     "${compact(concat(list(aws_security_group.bastion.id), split(",", "${var.security_group_ids}")))}"
   ]
   iam_instance_profile = "${var.iam_instance_profile}"
+  associate_public_ip_address = "${var.assign_public_ip}"
```

Comment:
Let's rename `assign_public_ip` into `associate_public_ip_address` to make it as similar to name as possible